src/app/reporting/ReportSchedulerImpl.h

```diff
@@ -33,29 +53,98 @@ class ReportSchedulerImpl : public ReportScheduler
-    // No action is needed by the ReportScheduler on ICD operation Mode changes
+
+    /**
+     * @brief When the ICD changes operation mode, no action is taken in this implementation.
+     */
     void OnICDModeChange() override{};
 
     // ReadHandlerObserver
+
+    /**
+     * @brief When a ReadHandler is added, adds a node and register it in the scheduler node pool. Scheduling the report here is
+     * un-necessary since the ReadHandler will call MoveToState(HandlerState::CanStartReporting);, which will call
+     * OnBecameReportable() and schedule the report.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnSubscriptionEstablished(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler becomes reportable, recalculate and reschedule the report.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnBecameReportable(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler report is sent, recalculate and reschedule the report.
+     *
+     * @note This method is called after the report is sent, so the ReadHandler is no longer reportable, and thus CanBeSynced and
+     * EngineRunScheduled of the node associated to the ReadHandler are set to false in this method.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnSubscriptionReportSent(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler is destroyed, remove the node from the scheduler node pool and cancel the timer associated to it.
+     */
     void OnReadHandlerDestroyed(ReadHandler * aReadHandler) override;
 
     virtual bool IsReportScheduled(ReadHandler * aReadHandler);
 
     void ReportTimerCallback() override;
 
 protected:
+    /**
+     * @brief Schedule a report for the ReadHandler associated to the node.
+     *
+     *  If a report is already scheduled for the ReadHandler, cancel it and schedule a new one.
+     *  If the timeout is 0, directly calls the TimerFired() method of the node instead of scheduling a report.
+     *
+     * @param[in] timeout The timeout to schedule the report.
+     * @param[in] node The node associated to the ReadHandler.
+     * @param[in] now The current system timestamp.
+     *
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, timer related error code otherwise (This can only fail on starting the timer)
+     */
     virtual CHIP_ERROR ScheduleReport(Timeout timeout, ReadHandlerNode * node, const Timestamp & now);
     void CancelReport(ReadHandler * aReadHandler);
     virtual void UnregisterAllHandlers();
 
 private:
     friend class chip::app::reporting::TestReportScheduler;
 
+    /**
+     * @brief Find the next timer when a report should be scheduled for a ReadHandler.
+     *
+     * @param[out] timeout The timeout to calculate.
```

Comment:
This is computing a delay from "now", right?  _That_ would be useful to document.